flink/v1.13/flink/src/main/java/org/apache/iceberg/flink/FlinkDynamicTableFactory.java

```diff
@@ -71,6 +72,12 @@ public class FlinkDynamicTableFactory implements DynamicTableSinkFactory, Dynami
           .noDefaultValue()
           .withDescription("Table name managed in the underlying iceberg catalog and database.");
 
+  private static final ConfigOption<Boolean> WRITE_UPSERT_ENABLED =
+      ConfigOptions.key("write-upsert-enabled")
+          .booleanType()
+          .defaultValue(false)
+          .withDescription("Whether to transform all INSERT/UPDATE_AFTER events to UPSERT.");
+
   // Flink 1.13.x change the return type from CatalogTable interface to ResolvedCatalogTable which extends the
   // CatalogTable. Here we use the dynamic method loading approach to avoid adding explicit CatalogTable or
   // ResolvedCatalogTable class into the iceberg-flink-runtime jar for compatibility purpose.
```

Comment:
Unrelated, but this dynamic function call can probably be removed now that we have code per major Flink version (unless this change was made in a patch release after 1.13.0).

I think @hililiwei has a PR that removes it iirc, but that PR also tackles much larger issues. @hililiwei can you possibly open a PR to just update the no-longer needed dynamic `getCatalogTable` call? I tried looking for the PR but couldn't find it.

Reply:
you mean here?
https://github.com/apache/iceberg/pull/4246#discussion_r817304711

https://github.com/apache/iceberg/pull/4246